src/include/cache.i

```diff
@@ -46,6 +46,19 @@ __wt_cache_read_gen_set(WT_SESSION_IMPL *session)
 }
 
 /*
+ * __wt_cache_read_gen_new --
+ *      Get the read generation for a new page in memory.
+ */
+static inline uint64_t
+__wt_cache_read_gen_new(WT_SESSION_IMPL *session)
+{
+	WT_CACHE *cache;
+
+	cache = S2C(session)->cache;
+	return (__wt_cache_read_gen(session) + cache->read_gen_oldest) / 2;
```

Comment:
The idea is for new pages to get a priority that puts them "in the middle" of existing pages.  That way, pages that are read once (e.g., in a big sequential scan) will not cause the top 50% of pages to be evicted from cache.

Of course, we don't have a global LRU queue, so we don't know the real oldest read generation.  This is just an attempt to calculate a reasonable value given our current data structures.